src/nti/fakestatsd/client.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function, absolute_import, division
+__docformat__ = "restructuredtext en"
+
+
+from perfmetrics.statsd import StatsdClient
+
+from .metric import Metric
+
+class FakeStatsDClient(StatsdClient):
+    """
+    A mock statsd client that tracks sent statsd metrics in memory
+    rather than pushing them over a socket.
+    """
+
+    def __init__(self, prefix=''):
+        """
+        Create a mock statsd client with the given prefix.
+        """
+        super(FakeStatsDClient, self).__init__(prefix=prefix)
+
+        # Monkey patch the socket to track things in memory instead
+        self._raw = _raw = []
+        self._metrics = _metrics = []
+
+        class DummySocket(object):
```

Comment:
I'm wondering, now, if we should capture the old value of `self.udp_sock` and provide this class with a `__getattr__` that forwards to that object? Is `udp_sock` considered public?

Reply:
So it's not documented, but it also is `_` prefixed. Do you imagine scenarios where you might still want to forward the metrics to an actual statsd server during testing?  I'm not convinced that makes a lot of sense to me. Perhaps we move this to a separate issue.